src/tests/class_tests/openms/source/NeighborSeq_test.cpp

```diff
@@ -0,0 +1,180 @@
+  double mass_2 = NeighborSeq::calculateMass("PPPPPPP");
+  double mass_3 = NeighborSeq::calculateMass("PEPEP");
+  // Compare the calculated masses with expected values
+  TEST_REAL_SIMILAR(mass_1, 799.815);
+  TEST_REAL_SIMILAR(mass_2, 697.814);
+  TEST_REAL_SIMILAR(mass_3, 567.579);
+}
+END_SECTION
+
+// Test section for the generateSpectrum function
+START_SECTION(MSSpectrum generateSpectrum(const String& peptide_sequence))
+{
+  MSSpectrum spec_1 = NeighborSeq::generateSpectrum("PEPT");
+  MSSpectrum spec_2 = NeighborSeq::generateSpectrum("PPP");
+  MSSpectrum spec_3 = NeighborSeq::generateSpectrum("AR");
+  
+  // Define the expected spectrum for "PEPT"
+  MSSpectrum spec_1_test;
+ 
+  Peak1D peak1;
+  peak1.setMZ(120.06552075);
+  peak1.setIntensity(1);
+  spec_1_test.push_back(peak1);
+
+  Peak1D peak2;
+  peak2.setMZ(217.11828498);
+  peak2.setIntensity(1);
+  spec_1_test.push_back(peak2);
+
+  Peak1D peak3;
+  peak3.setMZ(227.10263491);
+  peak3.setIntensity(1);
+  spec_1_test.push_back(peak3);
+
+  Peak1D peak4;
+  peak4.setMZ(324.15539914);
+  peak4.setIntensity(1);
+  spec_1_test.push_back(peak4);
+
+  Peak1D peak5;
+  peak5.setMZ(346.16087920);
+  peak5.setIntensity(1);
+  spec_1_test.push_back(peak5);
+
+// Define the expected spectrum for "PPP"
+  MSSpectrum spec_2_test;
+  Peak1D peak1;
+  peak1.setMZ(116.07060575);
+  peak1.setIntensity(1);
+  spec_2_test.push_back(peak1);
+
+  Peak1D peak2;
+  peak2.setMZ(195.11280491);
+  peak2.setIntensity(1);
+  spec_2_test.push_back(peak2);
+
+  Peak1D peak3;
+  peak3.setMZ(213.12336998);
+  peak3.setIntensity(1);
+  spec_3_test.push_back(peak3);
+
+// Define the expected spectrum for "AR"
+  MSSpectrum spec_3_test;
+  Peak1D peak1;
+  peak1.setMZ(175.11895291);
+  peak1.setIntensity(1);
+  spec_3_test.push_back(peak1);
+
+// Compare the generated spectra with the expected spectra
+  TEST_EQUAL(spec_1, spec_1_test);
+  TEST_EQUAL(spec_1, spec_2_test);
+  TEST_EQUAL(spec_1, spec_3_test);
+  TEST_EQUAL(spec_2, spec_1_test);
+  TEST_EQUAL(spec_2, spec_2_test);
+  TEST_EQUAL(spec_2, spec_3_test);
+  TEST_EQUAL(spec_3, spec_1_test);
+  TEST_EQUAL(spec_3, spec_2_test);
+  TEST_EQUAL(spec_3, spec_3_test);
+}
+END_SECTION
+
+// Test section for the compareSpectra function
+START_SECTION(bool compareSpectra(const MSSpectrum& spec1, const MSSpectrum& spec2, const double& ion_tolerance, const string& resolution))
+{
+  MSSpectrum spec1;
+  spec1.push_back(Peak1D(100.0, 1.0));
+  spec1.push_back(Peak1D(200.0, 1.0));
+  spec1.push_back(Peak1D(300.0, 1.0));
+  spec1.push_back(Peak1D(400.0, 1.0));
+
+
+  MSSpectrum spec2;
+  spec2.push_back(Peak1D(100.05, 1.0));
+  spec2.push_back(Peak1D(200.05, 1.0));
+  spec2.push_back(Peak1D(300.05, 1.0));
+  spec2.push_back(Peak1D(400.06, 1.0));
+
+
+  MSSpectrum spec3;
+  spec3.push_back(Peak1D(101.05, 1.0));
+  spec3.push_back(Peak1D(201.05, 1.0));
+  spec3.push_back(Peak1D(301.05, 1.0));
+  spec3.push_back(Peak1D(301.06, 1.0));
```

Comment:
in these spectra, either all peaks are shared, or none, right?
Can you add a test case with partially shared ions?
From a test perspective, it would be nice to have a separate function in the NeighborSeq class, which actually gives you the number of shared peaks. This would make testing more robust.